// knex/knex.js
// Knex.js
// --------------
//     (c) 2013-present Tim Griesser
//     Knex may be freely distributed under the MIT license.
//     For details and documentation:
//     http://knexjs.org

const knex = require('./lib/index');

/**
 * These export configurations enable JS and TS developers
 * to consume knex in whatever way best suits their needs.
 * Some examples of supported import syntax includes:
 * - `const knex = require('knex')`
 * - `const { knex } = require('knex')`
 * - `import * as knex from 'knex'`
 * - `import { knex } from 'knex'`
 * - `import knex from 'knex'`
 */
knex.knex = knex;
knex.default = knex;

// Use DATABASE_URL in production, otherwise default to local PostgreSQL settings
const dbConnection = process.env.DATABASE_URL
  ? {
      connectionString: process.env.DATABASE_URL, // Use DATABASE_URL in production
      ssl: { rejectUnauthorized: false }, // Enable SSL for managed databases like Render
    }
  : {
      // Fallback to local development settings
      host: 'localhost',
      port: 5432,
      user: 'postgres',
      password: '2002',
      database: 'ecommercedb',
    };

// Initialize knex with the appropriate configuration
module.exports = knex({
  client: 'pg', // PostgreSQL client
  connection: dbConnection, // Use dynamic connection based on environment
});
